inc/Integrations/Airtable/AirtableIntegration.php

```diff
@@ -18,6 +18,11 @@ public static function register_blocks(): void {
 
 		foreach ( $data_source_configs as $config ) {
 			$data_source = AirtableDataSource::from_array( $config );
+
+			if ( false === $config['service_config']['enable_blocks'] ?? true ) {
```

Comment:
Fun fact! Order of operations means that the comparison is done before the coalescing. Need parentheses here to avoid a bug:
```suggestion
			if ( false === ( $config['service_config']['enable_blocks'] ?? true ) ) {
```

https://www.php.net/manual/en/language.operators.precedence.php

Reply:
There actually isn't a bug, but the coalescing isn't necessary.

If it's `null` then the comparison is `false` and it will not `continue`, which is the desired behavior.
If it's `false`, then the coalescing won't change it, and it will go in and hit `continue`.
If it's `true` then the comparison is `false`, and still, it will not `continue`.

I'm not sure if it's better to use `array_key_exists` before getting the value or not...

Reply:
There's no logical bug, but there is a bug in intended behavior. The coalescing is necessary to prevent a warning when `$config['service_config']['enable_blocks']` is undefined (like the one [you reported](https://github.com/Automattic/remote-data-blocks/pull/322#issuecomment-2622838714) on this PR). In some environments, these warnings can be escalated to errors which effectively interrupt.

That's part of the semantic value of the `??` operator, because it swallows these warnings while being less verbose than `isset` or `array_key_exists`. But the order of operations matters.

Reply:
TIL, thank you!